tensorflow_probability/examples/jupyter_notebooks/Probabilistic_Principal_Component_Analysis.ipynb

```diff
@@ -0,0 +1,444 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Probabilistic Principal Component Analysis\n",
+    "\n",
+    "Probabilistic principal components analysis (PCA) is a dimensionality reduction technique that analyzes data via a lower dimensional latent space (Tipping & Bishop, 1999). It is often used when there are missing values in the data or for multidimensional scaling.\n",
+    "\n",
+    "This tutorial is a revision of the [Edward tutorial for Probabilistic PCA](http://edwardlib.org/tutorials/probabilistic-pca) using TensorFlow Probability stack, with an additional example of inference by Hamiltonian Monte Carlo."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import IPython\n",
+    "import matplotlib.pyplot as plt; plt.style.use('ggplot')\n",
+    "import numpy as np\n",
+    "import seaborn as sns; sns.set_context('notebook')\n",
+    "import tensorflow as tf\n",
+    "import tensorflow_probability as tfp\n",
+    "from tensorflow_probability import edward2 as ed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data\n",
+    "We use simulated data. We'll talk about the individual variables and what they stand for in the next section. For this example, each data point is 2-dimensional, $\\mathbf{x}_{n}\\in\\mathbb{R}^{2}$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def build_toy_dataset(N, D, K, sigma=1.0):\n",
```

Comment:
We prefer to generate synthetic data directly from the Edward model, rather than writing separate sampling code. (I realize you inherited this sin from the original notebook :). This guarantees we're using the same model for data generation and inference, and also saves a lot of code. See the PPCA-VI notebook (https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/jupyter_notebooks/Probabilistic_PCA.ipynb) for an example of doing this.